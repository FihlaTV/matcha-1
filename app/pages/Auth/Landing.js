import React, { Component } from 'react';
import { connect } from 'react-redux';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
import { injectIntl } from 'react-intl';
import { Actions } from 'react-native-router-flux';
import { selectCurrentUser } from 'app/pages/Auth/selector';
import type { UserType } from 'app/types/User';
import { FIELD_HEIGHT } from 'app/constants/design';
import Layout from 'app/components/Layout/Basic';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'space-around',
    height: 2 * FIELD_HEIGHT, // eslint-disable-line
  },
});

type Props = {
    currentUser: UserType,
    intl: any,
};

class Landing extends Component<Props> {
  componentWillMount() {
    if (this.props.currentUser !== undefined) {
      Actions.replace('feed');
    }
  }

  render() {
    return (
      <Layout noTabs>
        <View style={styles.container}>
          <Text>{this.props.intl.formatMessage({ id: 'LandingPage.welcome' })}</Text>
          <View>
            <TouchableOpacity onPress={() => Actions.login()}>
              <Text style={{ textAlign: 'center', textDecorationLine: 'underline' }}>
                {this.props.intl.formatMessage({ id: 'LoginPage.login' })}
              </Text>
            </TouchableOpacity>
          </View>
          <View>
            <TouchableOpacity onPress={() => Actions.signup()}>
              <Text style={{ textAlign: 'center', textDecorationLine: 'underline' }}>
                {this.props.intl.formatMessage({ id: 'SignupPage.signUp' })}
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </Layout>
    );
  }
}

const mapStateToProps = (state) => ({
  currentUser: selectCurrentUser(state),
});

export default connect(mapStateToProps)(injectIntl(Landing));
