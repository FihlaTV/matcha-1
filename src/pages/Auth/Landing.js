// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
import { injectIntl } from 'react-intl';
import { Actions } from 'react-native-router-flux';
import { selectCurrentUser } from 'src/pages/Auth/selector';
import { FIELD_HEIGHT } from 'src/constants/design';
import Layout from 'src/components/Layout/Basic';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'space-around',
    height: 2 * FIELD_HEIGHT, // eslint-disable-line
  },
});

type Props = {
    currentUser: UserType,
    intl: any,
};

class Landing extends Component<Props> {
  componentWillMount() {
    if (this.props.currentUser !== undefined) {
      Actions.replace('feed');
    }
  }

  render() {
    return (
      <Layout noTabs>
        <View style={styles.container}>
          <Text>{this.props.intl.formatMessage({ id: 'LandingPage.welcome' })}</Text>
          <View>
            <TouchableOpacity onPress={() => Actions.push('oauth')}>
              <Text style={{ textAlign: 'center', textDecorationLine: 'underline' }}>
                {this.props.intl.formatMessage({ id: 'LoginPage.login' })}
              </Text>
            </TouchableOpacity>
          </View>
          <View>
            <TouchableOpacity onPress={() => Actions.signup()}>
              <Text style={{ textAlign: 'center', textDecorationLine: 'underline' }}>
                {this.props.intl.formatMessage({ id: 'SignupPage.signUp' })}
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </Layout>
    );
  }
}

const mapStateToProps = (state) => ({
  currentUser: selectCurrentUser(state),
});

export default connect(mapStateToProps)(injectIntl(Landing));
