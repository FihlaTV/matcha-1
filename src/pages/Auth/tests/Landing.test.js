import React from 'react';
import Landing from 'src/pages/Auth/Landing';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Landing', () => {
  describe('render test', () => {
    it('should render', () => {
      const wrapper = shallow(<Landing />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
