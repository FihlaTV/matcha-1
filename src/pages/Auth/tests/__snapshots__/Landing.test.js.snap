// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing render test should render 1`] = `
<Landing
  store={Object {}}
/>
`;
